WordPress/Classes/Utility/WebKitViewController.swift

```diff
@@ -104,92 +124,128 @@ class WebKitViewController: UIViewController {
         if addsWPComReferrer {
             request.setValue("https://wordpress.com", forHTTPHeaderField: "Referer")
         }
-        webView.load(request)
+        DispatchQueue.main.async { [weak self] in
+            self?.webView.load(request)
```

Comment:
I'm curious, when was this being called from a secondary thread?

Reply:
It was happening every time when opening the `View Site` web view.

<img width="449" alt="screen shot 2018-06-07 at 12 52 46 pm" src="https://user-images.githubusercontent.com/9772967/41117292-3a770188-6a52-11e8-9ba5-0d8fefad5dc9.png">

It was called from `loadWebViewRequest` callback:
``` swift
authenticator.request(url: url, cookieJar: webView.configuration.websiteDataStore.httpCookieStore) { [weak self] (request) in
    self?.load(request: request)
}
```
That actually could be a better place to change thread.